### PR TITLE
Add rectification matrix in sdf, docs, and example usage

### PR DESCRIPTION
**Pull request by kjeppesen1**
Adds the ability to use a `rectification_matrix` sdf tag to populate the `camera_info_msg`, as discussed in this PR: https://github.com/ros-simulation/gazebo_ros_pkgs/pull/1409

### CODE REVIEW

**Comment by kjeppesen1** [comment]
@scpeters @jacobperron @ihasdapie 
Opening as a draft PR until @ihasdapie merges PR #1409

**Comment by scpeters** [comment]
> @scpeters @jacobperron @ihasdapie
> Opening as a draft PR until @ihasdapie merges PR #1409

@kjeppesen1  I just merged #1409, so can you resolve conflicts?

**Comment by kjeppesen1** [comment]
> > @scpeters @jacobperron @ihasdapie
> > Opening as a draft PR until @ihasdapie merges PR #1409
> 
> @kjeppesen1 I just merged #1409, so can you resolve conflicts?

Done. Marking this as ready to review now.

File: gazebo_plugins/src/gazebo_ros_camera.cpp [hunk 1]
```diff
@@ -435,15 +435,21 @@ void GazeboRosCamera::Load(gazebo::sensors::SensorPtr _sensor, sdf::ElementPtr _
...
+    stream >> rmat;
+    camera_info_msg.r[0] = rmat(0,0);
+    camera_info_msg.r[1] = rmat(0,1);
+    camera_info_msg.r[2] = rmat(0,2);
+    camera_info_msg.r[3] = rmat(1,0);
+    camera_info_msg.r[4] = rmat(1,1);
+    camera_info_msg.r[5] = rmat(1,2);
+    camera_info_msg.r[6] = rmat(2,0);
+    camera_info_msg.r[7] = rmat(2,1);
+    camera_info_msg.r[8] = rmat(2,2);
```
Comment (adityapande-1995) [inline gazebo_plugins/src/gazebo_ros_camera.cpp]: nit : The linter needs a space after the commas here : ``rmat(0, 0);``

Comment (kjeppesen1) [reply gazebo_plugins/src/gazebo_ros_camera.cpp]: done

File: gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.hpp [hunk 2]
```diff
@@ -70,6 +70,8 @@ class GazeboRosCameraPrivate;
       <!-- Projection matrix translation Tx, Ty between stereo cameras-->
       <Tx>240.5</Tx>
       <Ty>0</Ty>
+      <!-- Full 3x3 rectification matrix -->
```
Comment (jacobperron) [inline gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.hpp]: It might be worth documenting here that the values are in row-major order.

**Review by jacobperron — COMMENTED** [review]
One small comment, otherwise LGTM

File: gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.hpp (hunk 2 above)
Comment (kjeppesen1) [reply gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.hpp]: Thanks, added that to the comment.

**Review by jacobperron — APPROVED** [review]

**Comment by jacobperron** [comment]
@ros-pull-request-builder retest this please

**Pull request merged**